pkg/database/prepared_test.go

```diff
@@ -0,0 +1,139 @@
+package database_test
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/canonical/k8s-dqlite/pkg/database"
+)
+
+func TestPreparedPrepare(t *testing.T) {
+	const expectedQuery = "test query"
+
+	ctx := context.Background()
+	driver := &testDriver{
+		t: t,
+	}
+	db := database.NewPrepared(sql.OpenDB(&testConnector{driver: driver}))
+	defer db.Close()
+
+	stmt1, err := db.PrepareContext(ctx, expectedQuery)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stmt1.Close()
+
+	stmt2, err := db.PrepareContext(ctx, expectedQuery)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stmt2.Close()
+
+	if stmts := driver.stmts.Load(); stmts != 2 {
+		t.Errorf("invalid open statements: want 1, got %d", stmts)
+	}
+}
+
+func TestPreparedQuery(t *testing.T) {
+	ctx := context.Background()
+	driver := &testDriver{
+		t: t,
+	}
+	db := database.NewPrepared(sql.OpenDB(&testConnector{driver: driver}))
+	defer db.Close()
+
+	rows, err := db.QueryContext(ctx, "query 1")
+	if err != nil {
+		t.Error(err)
+	}
+	rows.Close()
+
+	if stmts := driver.stmts.Load(); stmts != 1 {
+		t.Errorf("unexpected number of open statements: want %d, got %d", 1, stmts)
```

Comment:
```suggestion
		t.Errorf("unexpected number of open statements: want 1, got %d", stmts)
```
Let's use consistent format across our tests.